src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -596,6 +622,90 @@ public override void WriteComponentAttribute(CodeRenderingContext context, Compo
         context.CodeWriter.WriteLine();
     }
 
+    private void WritePropertyAccess(CodeRenderingContext context, ComponentAttributeIntermediateNode node, ComponentIntermediateNode componentNode, string typeInferenceLocalName)
+    {
+        if (node?.TagHelper?.Name is null || node.Annotations["OriginalAttributeSpan"] is null)
+        {
+            return;
+        }
+
+        // Write the name of the property, for rename support. 
+        // __o = nameof(global::ComponentName.PropertyName);
+        var originalAttributeName = node.Annotations[ComponentMetadata.Common.OriginalAttributeName]?.ToString() ?? node.AttributeName;
+
+        int offset;
+        if (originalAttributeName == node.PropertyName)
+        {
+            offset = 0;
+        }
+        else if (originalAttributeName.StartsWith($"@bind-{node.PropertyName}", StringComparison.Ordinal))
+        {
+            offset = 5;
+        }
+        else
+        {
+            return;
+        }
+
+        var attributeSourceSpan = (SourceSpan)node.Annotations["OriginalAttributeSpan"];
+        attributeSourceSpan = new SourceSpan(attributeSourceSpan.FilePath, attributeSourceSpan.AbsoluteIndex + offset, attributeSourceSpan.LineIndex, attributeSourceSpan.CharacterIndex + offset, node.PropertyName.Length, attributeSourceSpan.LineCount, attributeSourceSpan.CharacterIndex + offset + node.PropertyName.Length);
+
+        context.CodeWriter.Write(DesignTimeVariable);
+        context.CodeWriter.Write(" = ");
+        context.CodeWriter.Write("nameof(");
```

Comment:
What happens when someone makes a local or other member in scope named `nameof`?

Reply:
We simply tell them that they should stop writing software, and take up farming, because farming has very little risk of a compiler dev reviewing their code and ruining their day.

Reply:
Why do we even need to use `nameof` here? Could we not just use a discard assignment?

Reply:
We don't have an instance to read the property from. I did originally have it as a property assignment (it was literally `((global::Foo)null).PropertyName = default;` so that it shows up in FAR as a `write` to match the semantics, but that got very annoying with generics. Now that I've changed the type inference bits though, we actually do have an instance, so that might work again.

Reply:
Okay this works, updated.